### PR TITLE
feat: added workflows to all actions

### LINKED ISSUE #16668: [CAL-4343] Add variable highlight and auto-complete everywhere (workflows)
This is already implemented for the email body of a workflow action see here: 

The text field for editing an SMS/Whatsapp workflow action and the text field for email subject don't use our Editor component. We don't want to allow editing the text but we want to have the highlight and auto-complete for variables

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-4343](https://linear.app/calcom/issue/CAL-4343/add-variable-highlight-and-auto-complete-everywhere-workflows)</sub>

### PR DESCRIPTION
**Pull request by krakenftw**
## What does this PR do?

- Fixes #16668 

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

### CODE REVIEW

**Comment by krakenftw** [comment]
@CarinaWolli i removed the normal text area and shifted everything to `Editor`
I might be missing something but please let me know how this looks.

File: packages/ui/components/editor/Editor.tsx [hunk 1]
```diff
@@ -74,19 +76,22 @@ export const Editor = (props: TextEditorProps) => {
...
+                setText={props.setText}
+                editable={editable}
+                excludedToolbarItems={props.excludedToolbarItems}
+                variables={props.variables}
+                updateTemplate={props.updateTemplate}
+                firstRender={props.firstRender}
+                setFirstRender={props.setFirstRender}
+                action={props.action}
+              />
+            )}
```
Comment (anikdhabal) [inline packages/ui/components/editor/Editor.tsx:94]: This is not good. Conditionally use a plugin for a particular use case. I think you can use the Editor's `excludedToolbarItems `property. We've used it in other components as well

Comment (krakenftw) [reply packages/ui/components/editor/Editor.tsx:94]: Are we talking about something like this?
```
excludedToolbarItems={
       step.action !== WorkflowActions.SMS_ATTENDEE && step.action !== WorkflowActions.SMS_NUMBER
           ? []
       : ["blockType", "bold", "italic"]
 }

Comment (krakenftw) [reply packages/ui/components/editor/Editor.tsx:94]: Also these conditions were there in `WorkflowStepContainer`.

File: packages/features/ee/workflows/components/WorkflowStepContainer.tsx [hunk 2]
```diff
@@ -914,63 +914,34 @@ export default function WorkflowStepContainer(props: WorkflowStepProps) {
                     {form.formState.errors.steps &&
                       form.formState?.errors?.steps[step.stepNumber - 1]?.emailSubject && (
                         <p className="mt-1 text-xs text-red-500">
-                          {form.formState?.errors?.steps[step.stepNumber - 1]?.emailSubject?.message || ""}
+                          {form.formState?.errors?.steps[step.stepNumber - 1]?.emailSubject?.message || ""}sd
```
Comment (anikdhabal) [inline packages/features/ee/workflows/components/WorkflowStepContainer.tsx:917]: Why this change?

File: packages/ui/components/editor/Editor.tsx (hunk 1 above)
Comment (anikdhabal) [reply packages/ui/components/editor/Editor.tsx:94]: > Are we talking about something like this?

You can try

File: packages/features/ee/workflows/components/WorkflowStepContainer.tsx (hunk 2 above)
Comment (krakenftw) [reply packages/features/ee/workflows/components/WorkflowStepContainer.tsx]: that was a typo.

File: packages/ui/components/editor/Editor.tsx [hunk 3]
```diff
@@ -74,19 +74,19 @@ export const Editor = (props: TextEditorProps) => {
...
+            <ToolbarPlugin
+              getText={props.getText}
+              setText={props.setText}
+              editable={editable}
+              excludedToolbarItems={props.excludedToolbarItems}
+              variables={props.variables}
+              updateTemplate={props.updateTemplate}
+              firstRender={props.firstRender}
+              setFirstRender={props.setFirstRender}
+            />
```
Comment (anikdhabal) [inline packages/ui/components/editor/Editor.tsx:89]: Revert the changes made in Editor component

**Review by anikdhabal — DISMISSED** [review]
LGTM!!

**Review by CarinaWolli — DISMISSED** [review]
Can we make it look same as we have it now? So 'Add variable' outside of the input field

That's how it should look:
Screenshot 2024-09-18 at 4 54 18 PM

**Comment by anikdhabal** [comment]
Screenshot 2024-10-01 183614

**Review by anikdhabal — APPROVED** [review]
